File: app/src/main/java/com/hititcs/dcs/view/home/view/HomePresenterImpl.java
package com.hititcs.dcs.view.home.view;

import javax.inject.Inject;

public class HomePresenterImpl implements HomeContract.HomePresenter {

  private HomeContract.HomeView view;

  @Inject
  public HomePresenterImpl(HomeContract.HomeView view) {
    this.view = view;
  }

  @Override public HomeContract.HomeView getView() {
    return view;
  }

  @Override public void setView(HomeContract.HomeView view) {
    this.view = view;
  }

  @Override public void dispose() {

  }
}

File: app/src/main/java/com/hititcs/dcs/view/home/view/HomePresenterImpl.kt
package com.hititcs.dcs.view.home.view

import javax.inject.Inject

class HomePresenterImpl @Inject constructor(private var view: HomeContract.HomeView) :
  HomeContract.HomePresenter {

  override fun getView(): HomeContract.HomeView {
    return view
  }

  override fun setView(view: HomeContract.HomeView) {
    this.view = view
  }

  override fun dispose() {
  }
}